### PR TITLE
[CPU] Optimize forked onednn binary size via removing gemm legacy above avx2.

### PR DESCRIPTION
**Pull request by luweizhou2016**
### Details:
 - *Fix RNN brgemm limitation*
 - *Fix reshape+FC limitation on inner product*
 - *Remove legacy gemm on ISAs above avx2*
 - *Enforce jit_gemm avx512 fall back jit_gemm avx2* 

### Tickets:
 - *CVS-118244*

### CODE REVIEW

**Comment by luweizhou2016** [comment]
ONEDNN PR https://github.com/openvinotoolkit/oneDNN/pull/219

**Comment by luweizhou2016** [comment]
Seems dien function regression hit CI test with ENALBE_LTO=ON.

File: src/plugins/intel_cpu/thirdparty/CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,8 @@ endif()
 
 if(ENABLE_LTO)
     set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_RELEASE ON)
+    #Only build gemm driver for AVX2 and below ISAs
+    set(ONEDNN_ENABLE_GEMM_KERNELS_ISA "AVX2" CACHE STRING "" FORCE)
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/thirdparty/CMakeLists.txt:17]: We need to disable avx512 gemm in any case, not only avx512 build

**Comment by dmitry-gorokhov** [comment]
@EgorDuplensky please review as well

**Comment by luweizhou2016** [comment]
Only have 3MB gain. Will not merge after dicussing.

**Pull request closed without merging**